### PR TITLE
Update the document to include the code formatter

### LINKED ISSUE #3166: Java Code Formatter
**Problem**
I was working on a Java ticket these days, and found that the format of our java codebase sometime is not consistent. 
Therefore, I am trying to propose a Java formatter for the team.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?
Knative developers

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.
- The formatter is easy to use for developers
- The CI tests should be configured as well. 
https://github.com/knative-sandbox/eventing-kafka-broker/blob/93b0ecb56d9550f7cff6f0089073e935fe2a3cd3/hack/update-codegen.sh#L80-L83

**Time Estimate (optional):** 
1

**Additional context (optional)**
The fotmatter I am proposing is called [Spotless](https://github.com/diffplug/spotless).

The Pros for choosing Spotless:
- Just run mvn spotless:apply does magic
- It is widely used, and actively been maintained!
- Can be integrated with CI testing
- It supports [many different](https://github.com/diffplug/spotless/tree/main/plugin-maven#google-java-format) formatting policy (google java style, eclipse jdt...)
- Support with IntelliJ IDEA and VSCode

### PR DESCRIPTION
**Pull request by Leo6Leo**
Fixes #3166 

## Proposed Changes

- Added the documentation for now to use the spotless formatter and go fmt

**Release Note**

```release-note
Added the documentation support for spotless and go fmt
```

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Review by Cali0707 — APPROVED** [review]
/lgtm

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -174,6 +176,31 @@ This step is required only if we're changing a protobuf definition file, or for
...
+go fmt ./...
+```
+
+### Java
+
+We are using Spotless to format our Java code. To format your code, run the following command within the `data-plane`
+directory:
+
+```shell
+mvn spotless:apply
```
Comment (pierDipi) [inline DEVELOPMENT.md:196]: Technically this is better because we don't require a maven installation
```suggestion
./mvnw spotless:apply
```

Comment (pierDipi) [reply DEVELOPMENT.md:196]: It's all good!

**Pull request merged**